okhttp/src/main/java/okhttp3/HttpUrl.kt

```diff
@@ -965,53 +962,49 @@ class HttpUrl internal constructor(builder: Builder) {
     }
 
     fun query(query: String?): Builder {
-      this.encodedQueryNamesAndValues = if (query != null) {
-        queryStringToNamesAndValues(canonicalize(query, QUERY_ENCODE_SET, alreadyEncoded = false,
-                strict = false, plusIsSpace = true, asciiOnly = true))
-      } else {
-        null
-      }
+      this.encodedQueryNamesAndValues = query?.canonicalize(
+          encodeSet = QUERY_ENCODE_SET,
+          plusIsSpace = true
+      )?.toQueryNamesAndValues()
```

Comment:
This is a nice example of how extension functions and the safe call operator work together.

Reply:
Yeah we traditionally have a lot of utils that accept null and early return that we can start removing. Close quietly comes to mind...